src/config.js

```diff
@@ -1,20 +1,120 @@
+ * @property  {String} senderEmail set sender's email or user only in case of supported services
+ * @property  {String} senderPassword set sender's password only in case of supported services
+ * @property  {Object} [auth]
+ * @property  {String} auth.senderEmail set sender's email or user only in case of custom service
+ * @property  {String} auth.senderPassword set sender's password only in case of custom service
+ * @property  {String} [host] - Host for your custom service
+ * @property  {Number} [port] - Port for your custom service
+ * @property  {Boolean} [secure] - Set to true for your custom service if you want a (TLS) secured connection
+ */
 
-const setConfig = (userConfig) => {
-  // TODO: Better  validation with joi
-  if (userConfig.senderEmail && userConfig.senderPassword) {
-    config = userConfig
+/**
+ * @private
+ * @type {config} configuration for mailer service
+ */
+let config = {}
+/**
+ * @exports 
+ * @param  {Object} userConfig set sender's email configuration
+ * @param  {URL} userConfig.hostingURL
+ * @param  {"Custom"|"126"|"163"|"1und1"|"AOL"|"DebugMail"|"DynectEmail"|"FastMail"|"GandiMail"|"Gmail"|"Godaddy"|"GodaddyAsia"|"GodaddyEurope"|"hot.ee"|"Hotmail"|"iCloud"|"mail.ee"|"Mail.ru"|"Maildev"|"Mailgun"|"Mailjet"|"Mailosaur"|"Mandrill"|"Naver"|"OpenMailBox"|"Outlook365"|"Postmark"|"QQ"|"QQex"|"SendCloud"|"SendGrid"|"SendinBlue"|"SendPulse"|"SES"|"SES-US-EAST-1"|"SES-US-WEST-2"|"SES-EU-WEST-1"|"Sparkpost"|"Yahoo"|"Yandex"|"Zoho"|"qiye.aliyun"} userConfig.mailService- Set mailing service to use defaults to gmail
+ * @param  {String} userConfig.senderEmail set sender's email in case of supported services or user in case of custom service
+ * @param  {String} userConfig.senderPassword set sender's password in case of supported services or user in case of custom service
+ * @param  {String} [userConfig.host] - Host for your custom service
+ * @param  {Number} [userConfig.port] - Port for your custom service
+ * @param  {Boolean} [userConfig.secure ]- Set to true for your custom service if you want a (TLS) secured connection
+ * @throws Invalid Configurations provided for custom service
+ * @throws Invalid settings provided for senderEmail, senderPasword, or mailService
+ * @returns {config} configuration that has been set
+ */
+function setConfig(
+  userConfig = {
+    hostingURL: '',
+    mailService: 'Gmail',
+    senderEmail: '',
+    senderPassword: '',
+    host: '',
+    port: 465,
+    secure: false,
+  }
+) {
+  config = {}
+  // config.hostingURL = (validator.isURL(userConfig.hostingURL || process.env.HOSTING_URL))?userConfig.hostingURL || process.env.HOSTING_URL: '';
+  config.hostingURL = userConfig.hostingURL || process.env.HOSTING_URL;
```

Comment:
Remove `|| process.env` and let the configuration only be set from the config object by the dev. It will also clean up the validation.

Reply:
I would recommend to be there, it makes the development job much easier with env variables.